File: baekjoon/c++/11659.cpp
#include <iostream>
using namespace std;

int dp[100000 + 1];
int numbers[100000 + 1];

int main(void){
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int len, case_cnt;
    cin >> len >> case_cnt;
    for (size_t i = 1 ; i <= len; ++i) {
        int temp;
        cin >> temp;
        numbers[i] = temp;
        dp[i] = dp[i - 1] + numbers[i];
    }

    for(size_t i = 1; i <= case_cnt; ++i) {
        int start,end;
        cin >> start >> end;
        cout << dp[end] - dp[start - 1] << '\n';
    }
}

File: baekjoon/python/11659.py
import sys
input = sys.stdin.read
data = input().split()

index = 0
len = int(data[index])
index += 1
case_cnt = int(data[index])
index += 1

numbers = [0] * (len + 1)
dp = [0] * (len + 1)
for i in range(1, len + 1):
    temp = int(data[index])
    index += 1
    numbers[i] = temp
    dp[i] = dp[i - 1] + numbers[i]
results = []
for i in range(case_cnt):
    start = int(data[index])
    index += 1
    end = int(data[index])
    index += 1
    results.append(dp[end] - dp[start - 1])
for result in results:
    print(result)